### PR TITLE
release: publish signed binaries instead of unsigned

### PR DESCRIPTION
**Pull request by rail**
Previously, `publish-provisional-artifacts`, when `-bless` is passed, tried to publish unsigned MacOS ARM64 binaries, while at that stage they are already signed.

This PR stops using `publish-provisional-artifacts` to publish `latest` archives and uses simple shell logic to copy archives.

Epic: none
Release note: None

### CODE REVIEW

File: pkg/release/build.go [hunk 1]
```diff
@@ -290,6 +292,8 @@ const (
 	PlatformMacOS
 	// PlatformMacOSArm is the Darwin aarch6 target.
 	PlatformMacOSArm
+	// PlatformMacOSArmSigned is the Darwin aarch6 target.
+	PlatformMacOSArmSigned
```
Comment (rickystewart) [inline pkg/release/build.go:296]: This is pretty confusing IMO. MacOSARMSigned and MacOSARM are not really different platforms at all. Instead maybe add a bool flag to `BuildOptions`? Or an extra bool argument to `SuffixFromPlatform`, given that it seems to be the only place where it matters.

Comment (rickystewart) [reply pkg/release/build.go:296]: That seems to make sense, but whatever makes it most maintainable / understandable for you.

**Review by rickystewart — APPROVED** [review]

**Comment by rail** [comment]
bors r=rickystewart

**Comment by rail** [comment]
bors retry

**Pull request merged**